Emby.Server.Implementations/Library/DefaultAuthenticationProvider.cs

```diff
@@ -95,117 +90,97 @@ private void ConvertPasswordFormat(User user)
-                passwordHash.Id = _cryptographyProvider.DefaultHashMethod;
-                passwordHash.Hash = GetHashedStringChangeAuth(newPassword, passwordHash);
-            }else if (newPassword != null)
-            {
-                passwordHash.Hash = GetHashedString(user, newPassword);
-            }
-
-            if (string.IsNullOrWhiteSpace(passwordHash.Hash))
-            {
-                throw new ArgumentNullException(nameof(passwordHash.Hash));
-            }
-
-            user.Password = passwordHash.ToString();
-
-            return Task.CompletedTask;
-        }
-
-        public string GetPasswordHash(User user)
-        {
-            return string.IsNullOrEmpty(user.Password)
-                ? GetEmptyHashedString(user)
-                : user.Password;
-        }
-
-        public string GetEmptyHashedString(User user)
-        {
-            return GetHashedString(user, string.Empty);
-        }
-
-        public string GetHashedStringChangeAuth(string NewPassword, PasswordHash passwordHash)
-        {
-            return BitConverter.ToString(_cryptographyProvider.ComputeHash(passwordHash.Id, Encoding.UTF8.GetBytes(NewPassword), passwordHash.SaltBytes)).Replace("-", string.Empty);
-        }
-
-        /// <summary>
-        /// Gets the hashed string.
-        /// </summary>
+        }
+
+        public Task<bool> HasPassword(User user)
+        {
+            var hasConfiguredPassword = !IsPasswordEmpty(user, GetPasswordHash(user));
+            return Task.FromResult(hasConfiguredPassword);
+        }
+
+        private bool IsPasswordEmpty(User user, string passwordHash)
+        {
+            return string.IsNullOrEmpty(passwordHash);
+        }
+
+        public Task ChangePassword(User user, string newPassword)
+        {
+            //string newPasswordHash = null;
+            ConvertPasswordFormat(user);
+            PasswordHash passwordHash = new PasswordHash(user.Password);
+            if (passwordHash.Id == "SHA1" && string.IsNullOrEmpty(passwordHash.Salt))
+            {
+                passwordHash.SaltBytes = _cryptographyProvider.GenerateSalt();
+                passwordHash.Salt = PasswordHash.ConvertToByteString(passwordHash.SaltBytes);
+                passwordHash.Id = _cryptographyProvider.DefaultHashMethod;
+                passwordHash.Hash = GetHashedStringChangeAuth(newPassword, passwordHash);
+            }else if (newPassword != null)
+            {
+                passwordHash.Hash = GetHashedString(user, newPassword);
+            }
+
+            if (string.IsNullOrWhiteSpace(passwordHash.Hash))
+            {
+                throw new ArgumentNullException(nameof(passwordHash.Hash));
+            }
+
+            user.Password = passwordHash.ToString();
+
+            return Task.CompletedTask;
+        }
+
+        public string GetPasswordHash(User user)
+        {
+            return user.Password;
+        }
+
+        public string GetEmptyHashedString(User user)

```

Comment:
huh, do we need this function now?

Reply:
not once I changed a few more references. So removing it.